File: jar-common/src/main/java/com/beancounter/common/model/Currency.java
package com.beancounter.common.model;

import javax.persistence.Entity;
import javax.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;
import lombok.ToString;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@ToString(onlyExplicitlyIncluded = true)
public class Currency {
  @NonNull
  @Id
  @ToString.Include
  private String code;
  private String name;
  private String symbol;

}

File: jar-common/src/main/java/com/beancounter/common/model/Currency.kt
package com.beancounter.common.model

import javax.persistence.Entity
import javax.persistence.Id

@Entity
/**
 * Persistent representation of a Currency.
 */
data class Currency(@Id var code: String, var name: String? = "Dollar", var symbol: String? = "$") {
    constructor(code: String) : this(code, "Dollar", "$")

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Currency

        if (code != other.code) return false

        return true
    }

    override fun hashCode(): Int {
        return code.hashCode()
    }

    override fun toString(): String {
        return code
    }
}
